### PR TITLE
feat(elasticloadbalancingv2): support TCP idle timeout for Network Load Balancer Listener

### LINKED ISSUE #31310: aws-elasticloadbalancingv2:  support tcp idle timeout for Network Load Balancer
### Describe the feature

AWS Network Load Balancer now supports configurable TCP idle timeout.

Announcement: https://aws.amazon.com/about-aws/whats-new/2024/09/aws-network-load-balancer-tcp-idle-timeout/

### Use Case

To set idle timeout for NLB.

### Proposed Solution
It is likely that it cannot be set at present, and we need to wait for CloudFormation support.
It is necessary to set `tcp.idle_timeout.seconds` in the listener's Attributes.

```sh
aws elbv2 modify-listener-attributes \
          --listener-arn arn:aws:elasticloadbalancing:us-east-1:000011112222:listener/network/NLBTest/123/123 \
          --attributes \
              Key=tcp.idle_timeout.seconds,Value=600 
```

https://aws.amazon.com/blogs/networking-and-content-delivery/introducing-nlb-tcp-configurable-idle-timeout/

However, there is no way to configure it using CloudFormation.
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-elasticloadbalancingv2-listener.html

I also tried adding it to the load balancer's attributes, like with ALB, but I couldn't do it.
image

### Other Information

_No response_

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

all

### Environment details (OS name and version, etc.)

all

### PR DESCRIPTION
**Pull request by mazyu36**
### Issue # (if applicable)

Closes #31310.

### Reason for this change
To set idle timeout for NLB.

AWS Network Load Balancer now supports configurable TCP idle timeout.

Announcement: https://aws.amazon.com/about-aws/whats-new/2024/09/aws-network-load-balancer-tcp-idle-timeout/

### Description of changes
* Add attributes to `BaseListener` class.
* Add `tcpIdleTimeout` property to `NetworkListener` class.

### Description of how you validated changes
Add unit tests and integ test.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/nlb/network-listener.ts [hunk 1]
```diff
@@ -222,6 +229,20 @@ export class NetworkListener extends BaseListener implements INetworkListener {
     if (props.defaultTargetGroups) {
       this.setDefaultAction(NetworkListenerAction.forward(props.defaultTargetGroups));
     }
+
+    if (props.tcpIdleTimeout !== undefined && !Token.isUnresolved(props.tcpIdleTimeout)) {
```
Comment (badmintoncryer) [inline packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/nlb/network-listener.ts:233]: I think it is better to handle the case where `tcpIdleTimeout` is smaller than 1 sec.

```suggestion
    if (props.tcpIdleTimeout !== undefined && !Token.isUnresolved(props.tcpIdleTimeout)) { 
     if (props.tcpIdleTimeout.toMilliseconds() < Duration.seconds(1).toMilliseconds()) {
       throw new Error(`\`tcpIdleTimeout\`` must be between 60 and 6000 seconds, got ${props.tcpIdleTimeout.toMilliseconds()} milliseconds.`)
     }
```

And could you please add unit test about this?

**Review by badmintoncryer — CHANGES_REQUESTED** [review]
Thank you for your PR! I just made one comment.

**Comment by mazyu36** [comment]
@badmintoncryer 
Thank you for the review!
I've added a validation and an unit test.

**Review by sumupitchayan — APPROVED** [review]

**Comment by sumupitchayan** [comment]
@mergifyio update

**Pull request merged**